x/mongo/driver/topology/errors.go

```diff
@@ -28,21 +32,28 @@ type ConnectionError struct {
 
 // Error implements the error interface.
 func (e ConnectionError) Error() string {
```

Comment:
Suggest adding a compile check for `ConnectionError`, I just noticed we don't do this:

```go
var _ error = ConnectionError{}
```